src/inmanta/protocol/ipc_light.py

```diff
@@ -35,11 +36,22 @@ class ConnectionLost(IPCException):
     pass
 
 
+ServerContext = typing.TypeVar("ServerContext")
+R = typing.TypeVar("R")
+
+
+class IPCMethod(abc.ABC, typing.Generic[R, ServerContext]):
```

Comment:
I would sway the two generic parameters, so that it's in-line with the signature of the method. So first the input arguments and then the return value.
```suggestion
class IPCMethod(abc.ABC, typing.Generic[ServerContext, R]):
```